src/schemaGenerator.ts

```diff
@@ -284,18 +284,21 @@ function addSchemaLevelResolveFunction(schema: GraphQLSchema, fn: GraphQLFieldRe
 }
 
 function getFieldsForType(type: GraphQLType): GraphQLFieldMap<any, any> {
-    if ((type instanceof GraphQLObjectType) ||
-        (type instanceof GraphQLInterfaceType)) {
-        return type.getFields();
-    } else {
-        return undefined;
-    }
+  if ((type instanceof GraphQLObjectType) ||
+    (type instanceof GraphQLInterfaceType)) {
+    return type.getFields();
+  } else {
+    return undefined;
+  }
 }
 
-function addResolveFunctionsToSchema(schema: GraphQLSchema, resolveFunctions: IResolvers) {
+function addResolveFunctionsToSchema(schema: GraphQLSchema, resolveFunctions: IResolvers, resolverValidationOptions: any) {
   Object.keys(resolveFunctions).forEach((typeName) => {
     const type = schema.getType(typeName);
     if (!type && typeName !== '__schema') {
+      if (resolverValidationOptions.allowResolversNotInSchema) {
+        return;
```

Comment:
This will skip all of the rest of the function - you might just want to wrap the `throw` in an `if` statement.